hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AliasCollector.java

```diff
@@ -53,4 +53,17 @@ public void visitQueryPartTableReference(QueryPartTableReference tableReference)
 	public void visitFunctionTableReference(FunctionTableReference tableReference) {
 		tableReferenceMap.put( tableReference.getIdentificationVariable(), tableReference );
 	}
+
+	@Override
+	public void visitTableGroupJoin(TableGroupJoin tableGroupJoin) {
+		final TableGroup joinedGroup = tableGroupJoin.getJoinedGroup();
+		if ( joinedGroup.isInitialized() ) {
+			// Only process already initialized table groups to avoid
+			// forced initialization of joined lazy table groups
+			joinedGroup.accept( this );
+		}
```

Comment:
Can you please move this check up to the `AbstractSqlAstWalker`?

Reply:
Sure, updated.